src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -184,20 +202,38 @@ internal sealed class DocumentAttributes : IChecksummedObject, IObjectWritable
             /// </summary>
             public bool IsGenerated { get; }
 
+            /// <summary>
+            /// True if the source code contained in the document is only used in design-time (e.g. for completion),
+            /// but is not passed to the compiler when the containing project is built.
+            /// </summary>
+            public bool DesignTimeOnly { get; }
+
+            /// <summary>
+            /// The LSP client name that should get the diagnostics produced by this document; any other source
+            /// will not show these diagnostics.  For example, razor uses this to exclude diagnostics from the error list
+            /// so that they can handle the final display.
+            /// If null, the diagnostics do not have this special handling.
+            /// </summary>
+            public string? DiagnosticsLspClientName { get; }
```

Comment:
This shouldn't be exposed in a public API at all -- this is a temporary workaround for an LSP issue and would be getting deleted as a part of the pull diagnostics work.

Reply:
Strictly speaking these don't need to be part of the API at all, since they're not used from DocumentInfo, I was just following the pattern for consistency. Easy enough to remove.